spec/lib/saml_idp/incoming_metadata_spec.rb

```diff
@@ -56,5 +74,12 @@ module SamlIdp
       metadata = SamlIdp::IncomingMetadata.new(metadata_4)
       expect(metadata.sign_authn_request).to eq(false)
     end
+
+    it 'should parse single logout url as array' do
+      metadata = SamlIdp::IncomingMetadata.new(metadata_with_slo)
+      expect(metadata.single_logout_services).to be_a(Array)
+      expect(metadata.single_logout_services.size).to eq(1)
+      expect(metadata.single_logout_services.first[:binding]).to eq("HTTP-Redirect")
```

Comment:
maybe check the other values within the Hash `location: ..., default: ..., response_location: ...` should also always be present, right?